frontend/src/metabase/common/components/DataPicker/components/DataPickerModal.tsx

```diff
@@ -244,6 +233,39 @@ export const DataPickerModal = ({
       onClose={onClose}
       onItemSelect={handleItemSelect}
       isLoadingTabs={isLoadingAvailableData}
+      searchExtraButtons={[filterButton]}
     />
   );
 };
+
+const FilterButton = ({
+  value,
+  onChange,
+  options,
+}: {
+  value: CollectionItemModel[];
+  onChange: (value: CollectionItemModel[]) => void;
+  options: FilterOption[];
+}) => {
+  return (
+    <Popover zIndex={1000}>
```

Comment:
Since the command palette has z-index 500, maybe 450?

I'll have to remember to remove this z-index as part of https://github.com/metabase/metabase/pull/49442